statsmodels/stats/oaxaca.py

```diff
@@ -0,0 +1,227 @@
+    
+    '3' is the column of which we want to explain or which indicates
+    the two groups. In this case, it is if you rent.
+    
+    >>> model = Oaxaca(df.endog, df.exog, 3, hasconst = False)
+    >>> model.two_fold()
+    >>> ******************************
+        Unexplained Effect: 27.94091
+        Explained Effect: 130.80954
+        Gap: 158.75044
+        ******************************
+    >>> model.three_fold()
+    >>> ******************************
+        Characteristic Effect: 321.74824
+        Coefficent Effect: 75.45371
+        Interaction Effect: -238.45151
+        Gap: 158.75044
+        ******************************
+    """
+    def __init__(self, endog, exog, bifurcate, hasconst = True, swap = True, cov_type = 'nonrobust', cov_kwds=None):
+            if str(type(exog)).find('pandas') != -1:
+                bifurcate = exog.columns.get_loc(bifurcate)
+                endog, exog = np.array(endog), np.array(exog)
+
+            bi_col = exog[:, bifurcate]
+            endog = np.column_stack((bi_col, endog))
+            bi = np.unique(bi_col)
+            
+            #split the data along the bifurcate axis, the issue is you need to delete it after you fit the model for the total model.
+            exog_f = exog[np.where(exog[:, bifurcate] == bi[0])]
+            exog_s = exog[np.where(exog[:, bifurcate] == bi[1])]
+            endog_f = endog[np.where(endog[:, 0] == bi[0])]
+            endog_s = endog[np.where(endog[:, 0] == bi[1])]
+            exog_f = np.delete(exog_f, bifurcate, axis = 1)
+            exog_s = np.delete(exog_s, bifurcate, axis = 1)
+            endog_f = endog_f[:,1]
+            endog_s = endog_s[:,1]
+            endog = endog[:,1]
+            
+            self.gap = endog_f.mean() - endog_s.mean() 
+            
+            if swap and self.gap < 0:
+                endog_f, endog_s = endog_s, endog_f
+                exog_f, exog_s = exog_s, exog_f
+                self.gap = endog_f.mean() - endog_s.mean()
+            
+            if hasconst == False:
+                exog_f = sm.add_constant(exog_f, prepend = False)
+                exog_s = sm.add_constant(exog_s, prepend = False)
+                exog = sm.add_constant(exog, prepend = False)
+            
+            self._t_model = sm.OLS(endog, exog).fit(cov_type = cov_type, cov_kwds = cov_kwds)
+            self._f_model = sm.OLS(endog_f, exog_f).fit(cov_type = cov_type, cov_kwds = cov_kwds)
+            self._s_model = sm.OLS(endog_s, exog_s).fit(cov_type = cov_type, cov_kwds = cov_kwds)
+            
+            self.exog_f_mean = np.mean(exog_f, axis = 0)
+            self.exog_s_mean = np.mean(exog_s, axis = 0)
+            self.t_params = np.delete(self._t_model.params, bifurcate)
+        
+    def three_fold(self):
+        """
+        Calculates the three-fold Oaxaca Blinder Decompositions
+
+        Parameters
+        ----------
+
+        None
+
+        Returns
+        -------
+
```

Comment:
No blank line here or between entries below

Reply:
Thanks for going through it. I'll make those updates.